### DISCUSSION TITLE
confidence score in RTTM

### DISCUSSION THREAD

**Discussion by gatorbonita** [pyannote/pyannote-audio#1636] [Q&A]
Hi Pyannote community, is it possible to get the confidence score from the pipeline when generating RTTM? If not, can anyone point out how to get the likelihood which indicates the segment is spoken by the speakers? By investing the source code, I feel it may be somewhere in clustering? Thanks.

```
from pyannote.audio import Pipeline
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization", use_auth_token="xxxxxx")

# apply the pipeline to an audio file
diarization = pipeline("recording.wav")

# dump the diarization output to disk using RTTM format
with open("audio.rttm", "w") as rttm:
	diarization.write_rttm(rttm)
```

**Comment by hbredin** [comment]
Out of the box, there is no such thing available in pyannote.

This is a tricky problem because the underlying algorithm is a combination of multiple parts, each of them having their own uncertainties, and some of them (e.g. the clustering part) don't even provide such a thing.
